spec/models/well_spec.rb

```diff
@@ -262,31 +262,184 @@
   end
 
   [
-    [1000, 10, 50, 50, 0, nil],
-    [1000, 10, 10, 10, 0, nil],
-    [1000, 10, 20, 10, 0, 10],
```

Comment:
Have you lost this third scenario, where the 'current volume' (of the source well) is 10? Does the current volume of the source plate get used in the calculation? Or is it the measured volume that is checked?